packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -314,7 +314,7 @@ function useFocusContainment(scopeRef: RefObject<Element[]>, contain?: boolean)
 
     // Handle the Tab key to contain focus within the scope
     let onKeyDown = (e) => {
-      if (e.key !== 'Tab' || e.altKey || e.ctrlKey || e.metaKey || !shouldContainFocus(scopeRef)) {
+      if (e.key !== 'Tab' || e.altKey || e.ctrlKey || e.metaKey || !shouldContainFocus(scopeRef) || e.isComposing) {
```

Comment:
I see the same pattern in `useRestoreFocus` down below, can we verify if that one also needs the fix?

Reply:
good call, just checked and yes we'll need to add it to `useRestoreFocus`'s keydown handler as well otherwise hitting tab will move focus outside of a non-containing + `restoreFocus=true` popover that has an input inside it when the IME dropdown is open.